ocaml/xapi/xapi_vdi.ml

```diff
@@ -517,7 +519,8 @@ let destroy ~__context ~self =
       List.iter (fun vbd ->
           Helpers.log_exn_continue (Printf.sprintf "destroying VBD: %s" (Ref.string_of vbd))
             (fun vbd -> Db.VBD.destroy ~__context ~self:vbd) vbd) vbds;
-      (* Db.VDI.destroy ~__context ~self *)
+      (* If VDI destroyed is suspend VDI of VM then set the suspend_VDI field as null ref *)
+      List.iter (fun self -> Db.VM.set_suspend_VDI ~__context ~self ~value:Ref.null) suspend_vdi_vms;
```

Comment:
Instead of updating this relationship explicitly this way, it ought to work if you just add the relevant relation to the `all_relations` list in `datamodel.ml`
I think it would be
`(_vdi, "suspend_VDI_of"), (_vm, "suspend_VDI");`
and you would also need to add a field to the vdi definition: something like
`field ~qualifier:DynamicRO ~ty:(Ref _vm) "suspend_VDI_of" "If any, the VM whose suspended image is stored on this VDI. (Only has meaning if the VM is currently suspended)";`